src/js/tech/tech.js

```diff
@@ -218,12 +218,13 @@ class Tech extends Component {
       return;
     }
 
+    let player = this.player();
     let textTracksChanges = function() {
       let updateDisplay = Fn.bind(this, function() {
-        this.trigger('texttrackchange');
+        player.trigger('texttrackchange');
```

Comment:
Yep, your EDIT is what when the issue occurs.
Also, I see now what is trying to happen. I went down the rabbithole of only listening to the text tracks events in the display, but that doesn't work anymore because techs own the text tracks now. I might keep the listener in there and then have the tech tell the display to update its listeners.